barrels/LogMonkey/Logger.mc

```diff
@@ -0,0 +1,172 @@
+        }
+
+        (:release)
+        private function formLogMessage(tag, message) {
+        }
+
+        //! Handles the printing of a log message to the log file
+        //! @param [Toybox::Lang::String] tag The tag to apply to the log message
+        //! @param [Toybox::Lang::String] message The message to log
+        (:debug)
+        private function log(tag, message) {
+            // Print the log message
+            mLogStream.println(formLogMessage(tag, message));
+        }
+
+        (:release)
+        private function log(tag, message) {
+        }
+
+        //! Returns the type string of the given variable. Note, this function
+        //! will currently only return names within the Toybox.Lang module. If the
+        //! given variable isn't a type defined there then "Object" will be returned.
+        //! @param [Toybox::Lang::Object] variable The variable to get the type of
+        //! @return [Toybox::Lang::String] The name of the type of the given variable
+        (:debug)
+        private function getVariableType(variable) {
+            // If the given value is null we can't switch on it so perform
+            // a null check here. The return value should just be "null".
+            if (variable == null) {
+                return "null";
+            }
+
+            // Switch on the type of the variable and return the variable's
+            // class name accordingly.
+            switch (variable) {
+                case instanceof Lang.Array:
+                    return "Array";
+                case instanceof Lang.Boolean:
+                    return "Boolean";
+                case instanceof Lang.Char:
+                    return "Char";
+                case instanceof Lang.Dictionary:
+                    return "Dictionary";
+                case instanceof Lang.Double:
+                    return "Double";
+                case instanceof Lang.Exception:
+                    return "Exception";
+                case instanceof Lang.Float:
+                    return "Float";
+                case instanceof Lang.Long:
+                    return "Long";
+                case instanceof Lang.Method:
+                    return "Method";
+                case instanceof Lang.Number:
+                    return "Number";
+                case instanceof Lang.String:
+                    return "String";
+                case instanceof Lang.Symbol:
+                    return "Symbol";
+                case instanceof Lang.WeakReference:
+                    return "WeakReference";
+                default:
+                    return "Object";
+            }
+        }
+
+        (:release)
+        private function getVariableType(variable) {
+        }
+
+        //! Log the given message under the given tag
+        //! @param tag The tag to apply to the log message
+        //! @param message The message to log
+        (:debug)
+        public function logMessage(tag, message) {
```

Comment:
All MonkeyC code I've seen to date does not explicitly label public functions as such. Should this code reflect existing convention, or is it starting a revolution?

Reply:
I think with the introduction of public/private/protected we should start using them as a coding convention even if it's not grammatically required but I'm also not the authority to dictate this. Unless there's a strong opinion against it I'm going to leave them in.

Reply:
I don't mind. I only bring it up because this is just a Java convention that is sneaking into non-Java code. Ideally all of the code in this repo would follow a consistent style, but I'm not convinced that this is one of those style things that really matters.

Reply:
I agree it is a bit more Java-y but so is switching the grammar from hidden/<nothing> to private/protected/public. If we chose to go that route with the grammar I think it may make sense to update our teams convention approach. I'll update the patch to remove the public definitions and leave this open to debate internally.